.travis.yml

```diff
@@ -1,26 +1,59 @@
 language: cpp
 
-services:
-  - docker
-
-cache:
-  ccache: true
-
 env:
   global:
     - CCACHE_DIR=$HOME/.ccache
     - PX4_DOCKER_REPO=px4io/px4-dev-ros:2018-04-22
-    - BUILD_WITH_CMAKE = "mkdir Build; cd Build; cmake ..; make"
-    - BUILD_WITH_CATKIN = "source /opt/ros/kinetic/setup.bash; mkdir -p ~/catkin_ws/src; cd ~/catkin_ws/src; catkin_init_workspace; ln -s ${TRAVIS_BUILD_DIR} .; cd ~/catkin_ws; catkin_make"
-    - VALIDATE_SDF_SCHEMAS = "source ./scripts/validate_sdf.bash"
 
-  matrix:
-    # Cmake build
-    - STEP=${BUILD_WITH_CMAKE}
-    # Catkin build
-    - STEP=${BUILD_WITH_CATKIN}
-    # Validate SDF schemas
-    - STEP=${VALIDATE_SDF_SCHEMAS}
+matrix:
+    include:
+        - os: linux
+          services:
+            - docker
+          cache:
+            ccache: true
+          env: # Cmake build
+            - BUILD="mkdir Build; cd Build; cmake ..; make"
+        - os: linux
+          services:
+            - docker
+          cache:
+            ccache: true
+          env: # Catkin build
+            - BUILD="source /opt/ros/kinetic/setup.bash; mkdir -p ~/catkin_ws/src; cd ~/catkin_ws/src; catkin_init_workspace; ln -s ${TRAVIS_BUILD_DIR} .; cd ~/catkin_ws; catkin_make"
+        - os: linux
+          services:
+            - docker
+          cache:
+            ccache: true
+          env: # Validate SDF schemas
+            - BUILD="source ./scripts/validate_sdf.bash"
+        - os: osx
+          osx_image: xcode9.3
+          cache:
+            ccache: true
+
+before_install:
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
+    brew update;
+    git clone --depth 1 https://github.com/mavlink/c_library_v2.git /usr/local/include/mavlink/v2.0;
+    rm -rf /usr/local/include/mavlink/v2.0/.git;
+    fi
+
+install:
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
+    sudo -H pip install --upgrade pip setuptools;
+    sudo -H pip install nose tornado rospkg jinja2;
+    brew install eigen graphviz protobuf opencv3;
```

Comment:
Can we use the px4-dev and px4-sim brews here? That would be a good test for everything.
@LorenzMeier

Reply:
https://dev.px4.io/en/setup/dev_env_mac.html